src/main/java/jek/collections/IntArrayList.java

```diff
@@ -0,0 +1,296 @@
+package jek.collections;
+
+import java.util.*;
+
+public class IntArrayList implements List {
+
+    private int end;
+    private Object[] ial;
+
+    public IntArrayList() {
+        ial = new Object[10];
+        end = 0;
+    }
+
+    @Override
+    public int size() {
+
+        return end;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return (end == 0);
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        for (Object search : ial)
+            if (search.equals(o))
+                return true;
+
+        return false;
+    }
+
+    /* вот запилил итератор, но еще не особо прочекал его, бо уже иду спать(
+    * я еще перепроверю, и то шо ты в комментах написал тож поправлю.
+    * но наверн уже на выхах, бо в чтв и птнц дохуя работаю.
+    * пис.
+    *
+    *
+    *  */
+
+    @Override
+    public Iterator iterator() {
```

Comment:
While working on Iterators please pay attention to it's documentation from that perspective on how it should behave when somebody already got the iterator and then the list was changed.